### ISSUE TITLE
william: implementar funções de perda de classificação

### ISSUE THREAD

**Issue by gabriel-ferreira-da-silva** [gabriel-ferreira-da-silva/thoth#7]
implementar as seguintes funções de perda para classifificação
- Log-Likelihood Loss
- Binary Cross-Entropy Loss
- Softmax
- cross entropy loss
![Image](https://github.com/user-attachments/assets/845ef112-75b2-4b5e-9930-569313dc62f5)

![Image](https://github.com/user-attachments/assets/0e0f5677-7e37-4197-845e-cb78b3560ef9)

![Image](https://github.com/user-attachments/assets/15da122c-aeec-44a7-8bf3-7b05da67366a)

![Image](https://github.com/user-attachments/assets/c7a3a96b-790a-49a6-bd30-dc5bc51e71d8)
